### PR TITLE
fix genesis exporting of tendermint metadata

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

closes: #191 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ### Bug Fixes
 
+* (07-tendermint) [#\210](https://github.com/cosmos/ibc-go/pull/210) Correctly exported consensus metadata on genesis restarts for tendermint clients.
```
Comment (colin-axner) [inline CHANGELOG.md:41]:
```suggestion
* (07-tendermint) [#\210](https://github.com/cosmos/ibc-go/pull/210) Exporte all consensus metadata on genesis restarts for tendermint clients.
```

File: modules/light-clients/07-tendermint/types/genesis_test.go [hunk 2]
```diff
@@ -1,38 +1,89 @@
...
 	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
-	commitmenttypes "github.com/cosmos/ibc-go/modules/core/23-commitment/types"
 	"github.com/cosmos/ibc-go/modules/light-clients/07-tendermint/types"
+	ibctesting "github.com/cosmos/ibc-go/testing"
 )
 
+// expected export ordering:
+// processed height and processed time per height
+// then all iteration keys
 func (suite *TendermintTestSuite) TestExportMetadata() {
```
Comment (colin-axner) [inline modules/light-clients/07-tendermint/types/genesis_test.go:14]: This test could be better using a for loop for the updates, but it doesn't feel worth the effort. I think this covers the necessary purposes

File: modules/light-clients/07-tendermint/types/store.go [hunk 3]
```diff
@@ -82,20 +82,37 @@ func deleteConsensusState(clientStore sdk.KVStore, height exported.Height) {
...
+func IterateConsensusMetadata(store sdk.KVStore, cb func(key, val []byte) bool) {
 	iterator := sdk.KVStorePrefixIterator(store, []byte(host.KeyConsensusStatePrefix))
 
+	// iterate over processed time and processed height
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
 		keySplit := strings.Split(string(iterator.Key()), "/")
 		// processed time key in prefix store has format: "consensusState/<height>/processedTime"
-		if len(keySplit) != 3 || keySplit[2] != "processedTime" {
+		if len(keySplit) != 3 {
```
Comment (AdityaSripal) [inline modules/light-clients/07-tendermint/types/store.go:95]: This will not work since the iteration keys have a completely different format

**Review by AdityaSripal — CHANGES_REQUESTED** [review]
This solution does not work for iteration keys

File: modules/light-clients/07-tendermint/types/store.go [hunk 4]
```diff
@@ -82,20 +82,37 @@ func deleteConsensusState(clientStore sdk.KVStore, height exported.Height) {
...
+		if cb(iterator.Key(), iterator.Value()) {
+			break
 		}
+	}
 
+	// iterate over iteration keys
+	iterator = sdk.KVStorePrefixIterator(store, []byte(KeyIterateConsensusStatePrefix))
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
```
Comment (AdityaSripal) [inline modules/light-clients/07-tendermint/types/store.go:115]: Ahh I see this, good work!

**Review by AdityaSripal — APPROVED** [review]

**Pull request merged**